libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2838,16 +2845,24 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 				"Using \"." + memberName + "(...)\" is deprecated. Use \"{" + memberName + ": ...}\" instead."
 			);
 
-		if (
-			funType->kind() == FunctionType::Kind::ArrayPush &&
-			arguments.value().numArguments() != 0 &&
-			exprType->containsNestedMapping()
-		)
-			m_errorReporter.typeError(
-				8871_error,
-				_memberAccess.location(),
-				"Storage arrays with nested mappings do not support .push(<arg>)."
-			);
+		if (exprType->containsNestedMapping())
```

Comment:
I'm not sure if this should be the first condition, becuase that would throw if the type is not nameable:

```
contract C {
    function f() external {
        C.f();
    }
}
```

```
: Internal compiler error during compilation:
: /solidity/libsolidity/ast/Types.h(278): Throw in function virtual bool solidity::frontend::Type::containsNestedMapping() const
: Dynamic exception type: boost::wrapexcept<solidity::langutil::InternalCompilerError>
: std::exception::what: Called for a non nameable type.
: [solidity::util::tag_comment*] = Called for a non nameable type.
```

Maybe check in here if `kind` is `ArrayPop` or `ArrayPush`? That should fix this problem.

Reply:
Meh. I think my original solution was the cleanest here.. reverting to it..